app/src/main/java/edu/grinnell/appdev/grinnelldirectory/Activity/MainActivity.java

```diff
@@ -1,15 +1,169 @@
 package edu.grinnell.appdev.grinnelldirectory.Activity;
 
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
 
+import edu.grinnell.appdev.grinnelldirectory.APICaller;
+import edu.grinnell.appdev.grinnelldirectory.Interfaces.APICallerInterface;
+import edu.grinnell.appdev.grinnelldirectory.Model.Person;
 import edu.grinnell.appdev.grinnelldirectory.R;
+import edu.grinnell.appdev.grinnelldirectory.Model.User;
+
+
+public class MainActivity extends AppCompatActivity implements APICallerInterface {
+
+
+    private APICaller apiCaller;
+    private User user;
 
-public class MainActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        user = new User("test1stu", "selfserv1");
+        apiCaller = new APICaller(user, this);
+
+
+        runSuccessfulTests();
+        runFailingTests();
+    }
+
+    public void runSuccessfulTests() {
```

Comment:
Let's take all the testing out of MainActivity. We can create a different activity for that